openedx/core/djangoapps/util/legacy_ip.py

```diff
@@ -0,0 +1,47 @@
+"""
+Utilities for migrating to safer IP address determination.
+
+This module used to contain utilities for reading the IP addresses of
+a request, but those have since moved ``edx_django_utils.ip``.
+
+What remains are the "legacy IP" utils, which should be used only
+temporarily when switching a piece of code from using the leftmost IP
+(legacy IP) to using the safest IP or full public IP chain (using
+edx-django-utils).
+"""
+
+from edx_toggles.toggles import WaffleSwitch
+
+
+# .. toggle_name: ip.legacy
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: Emergency switch to revert to use the older, less secure method for
+#   IP determination. When enabled, instructs switch's callers to revert to using the *leftmost*
+#   IP from the X-Forwarded-For header. When disabled (the default), callers should use the new
+#   code path for IP determination, which has callers retrieve the entire external chain or pick
+#   the leftmost or rightmost IP from it. The construction of the external chain is configurable
+#   via ``CLOSEST_CLIENT_IP_FROM_HEADERS``.
+#     This toggle, as well as any other legacy IP references, should be deleted (in the off
+#   position) when the new IP code is well-tested and all IP-reliant code has been switched over.
+# .. toggle_warning: This switch does not control the behavior of this module. Callers must
+#   opt into querying this switch, and can call ``get_legacy_ip`` if the switch is enabled.
+# .. toggle_use_cases: temporary
+# .. toggle_creation_date: 2022-03-24
+# .. toggle_target_removal_date: 2023-01-01
```

Comment:
[proposal] We may want to follow improvements being made to DEPR that help better tie these dates to appropriate named releases. If you have any thoughts on this, you could mention something like the following in the warning:
```
This is being introduced in the Olive named releases, and may be removed as early as the following named release.
```
Whatever you decide is fine.

Reply:
Hmm... I don't know if it's needed for those one. The toggle only controls code that opts into using the legacy-ip code, and my plan was actually to remove all those call locations now that we've seen that the new IP calculation isn't causing any issues.

Reply:
I do need to update the toggle's warning annotation, though, to account for the change in where code lives.

Reply:
Got it. You might mention that this could be removed as soon as all references have been removed. (Although that might seem obvious, it would clarify to people like me that it doesn't require waiting for a different named release.)

Reply:
I think the toggle description already covers that. Pushing some general tweaks, let me know what you think,